src/Rest.php

```diff
@@ -0,0 +1,159 @@
+<?php
+/**
+ * Rest endpoints and functionality
+ *
+ * @package Bylines
+ */
+
+namespace Bylines;
+
+use Bylines\Objects\Byline;
+
+/**
+ * Rest endpoints and functionality
+ */
+class Rest {
+
+	/**
+	 * Register meta
+	 */
+	public static function register_meta() {
+		foreach ( Content_Model::get_byline_supported_post_types() as $post_type ) {
+			register_meta(
+				'post',
+				'bylines',
+				array(
+					'object_subtype' => $post_type,
+					'single'         => true,
+					'type'           => 'array',
+					'show_in_rest'   => array(
+						'schema' => array(
+							'type'  => 'array',
+							'items' => array(
+								'type' => 'string',
+							),
+						),
+					),
+				)
+			);
+		}
+	}
+
+	/**
+	 * Instead of writing loads of custom logic for the rest api
+	 * and the block editor, we "fake" a metadata that contains
+	 * the bylines and then use that during the save to set the
+	 * byline terms.
+	 *
+	 * @param mixed  $value     The value to return, either a single metadata value or an array
+	 *                          of values depending on the value of `$single`. Default null.
+	 * @param int    $object_id ID of the object metadata is for.
+	 * @param string $meta_key  Metadata key.
+	 */
+	public static function filter_meta( $value, $object_id, $meta_key ) {
+		if ( 'bylines' !== $meta_key ) {
+			return $value;
+		}
+		$byline_display_data = array();
+		foreach ( get_bylines( $object_id ) as $byline ) {
+			$display_name          = $byline->display_name;
+			$term                  = is_a( $byline, 'WP_User' ) ? 'u' . $byline->ID : $byline->term_id;
+			$byline_display_data[] = (string) $term;
+		}
+		return array( $byline_display_data );
+	}
+
+	/**
+	 * Register the bylines rest api route.
+	 */
+	public static function register_route(): void {
+		register_rest_route(
+			'bylines/v1',
+			'/bylines',
+			array(
+				array(
+					'methods'             => 'GET',
+					'callback'            => function( $request ) {
+						$bylines = Admin_Ajax::get_possible_bylines_for_search( (string) $request['s'] );
+						return rest_ensure_response( $bylines );
+					},
+					'permission_callback' => function() {
+						return current_user_can( get_taxonomy( 'byline' )->cap->assign_terms );
+					},
+					'args'                => array(
```

Comment:
we probably need args for include here so that the editor can pass the selected bylines to the search/list endpoint so we're sure they're always available in the edit/render context.